config/clang/clang-common.sh

```diff
@@ -1,19 +1,39 @@
-#!/bin/env bash
-# Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+#!/bin/bash
+# Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
 # For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
-set -eo pipefail
+set -euo pipefail
+#set -o posix
+LC_CTYPE="C"
+
+
+if [ -z "${L_GIT_DIR+unbound}" ] && [ -z "$(global L_GIT_DIR="$(git rev-parse --show-toplevel 2>/dev/null)")" ]; then
+  global L_GIT_DIR="$(readlink -f "$(dirname "$0")/../..")"
+fi
+
+. "${L_GIT_DIR}/tools/version_check.sh"
+
+_extract_number () { echo "${1}" | sed -e "s|[^[:digit:]]*\([[:digit:]]\+\)|\1|"; }
+_do_search () { find "${2}" -name "${1}-[[:digit:]][[:digit:]]*"; }
+
+_get_latest () {
+  latest=${1}
+  set -f; IFS=":";
+  for p in $PATH; do
+    for file in $(_do_search "${1}" "${p}"); do
+      if [ "$(expr "$(_extract_number "${latest}")" ">" "$(_extract_number "${file}")")" ]; then
```

Comment:
The `for` loop iterate the entries of `$PATH` by setting the field splitting character to `:` which is the character used to deliminate paths in `$PATH`.  `set -f` is there to solve a corner case of a blank entry from resolving to the current directory via pathname expansion.  Both of these are automatically reset since functions are invoked as a subshell.

So to answer your question, `_do_search` is invoked for each path stored in `$PATH`.